charts/k8ssandra/templates/cassandra/cassdc.yaml

```diff
@@ -66,9 +66,19 @@ spec:
       credentials_validity_in_ms: {{ .Values.cassandra.auth.cacheValidityPeriodMillis }}
       credentials_update_interval_in_ms: {{ .Values.cassandra.auth.cacheUpdateIntervalMillis }}
     jvm-options:
-      initial_heap_size: "800M"
-      max_heap_size: "800M"
-
+{{- if (index .Values.cassandra.datacenters 0).heap }}
+      initial_heap_size: {{ default "800M" (index .Values.cassandra.datacenters 0).heap.size }}
+      max_heap_size: {{ default "800M" (index .Values.cassandra.datacenters 0).heap.size }}
+      heap_size_young_generation: {{ default "1.6Gi" (index .Values.cassandra.datacenters 0).heap.newGenSize }}
+{{- else if .Values.cassandra.heap }}
```

Comment:
If we define defaults in `k8ssandra/values.yaml` for `cassandra.heap.size` and for `cassandra.heap.newGenSize`, then lines 73 - 79 can be simplified to an `else` block that simply assigns those values. Note that even if the user does not set these properties, then the chart will inherit the values defined in `values.yaml`.

Reply:
Adding this block to `values.yaml` under `cassandra`.
```yaml
heap:
    size: 800M
    newGenSize: 400M
```

Reply:
@burmanm made a good point earlier in a separate review that we may not want to explicitly set these. I agree. I do think we should have commented out and documented though in `values.yaml` to make it more clear what config options are available. What do you think?

Reply:
I agree, it may be more reasonable to offer the customization options as opposed to explicitly providing/setting any values.

I will insert the comments into the values.yaml for the `cluster` and `dc` level settings available.